app/code/core/Mage/Adminhtml/controllers/Promo/CatalogController.php

```diff
@@ -308,6 +308,38 @@ public function applyRulesAction()
         $this->_redirect('*/*');
     }
 
+     /**
+     * Duplicate catalog rule and edit
+     */
+    public function duplicateAction()
+    {
+        $errorMessage = Mage::helper('catalogrule')->__('Cannot copy rule ');
+        $params = $this->getRequest()->getParams();
+        try {
+            if($model = Mage::getModel('catalogrule/rule')->load($params['id'])) {
+                $_currentrule = $model->getData();
+                $_newrule = Mage::getModel('catalogrule/rule');
+                $_newrule->setData($_currentrule);
+                unset($_newrule['rule_id']);
+                $_newrule['is_active'] = 0;
+                $_newrule['name'] = $_currentrule['name'] . " (Copy)";
+                $_newrule['identifier'] = $_currentrule['identifier'] . "-copy";
+                unset($_currentrule);
+                $_newrule->save();
```

Comment:
The `$_currentrule` and `$_newrule` variable names are not consistent with Magento conventions of using camelCase and not adding underscore prefixes to variable names (these serve no purpose inside class methods).

While `Varien_Object` does support ArrayAccess it's use is very limited by convention so I would prefer to see the getters/setters used instead.